event.go

```diff
@@ -267,24 +267,39 @@ func (p *propertyParser) getPropertyName(i int) string {
 // parsing property value data pointer should point to the right memory.
 // eventParses controls data pointer with each GetPropertyValue call.
 func (p *propertyParser) getPropertyValue(i int) (interface{}, error) {
-	if int(C.PropertyIsStruct(p.info, C.int(i))) == 1 {
-		return p.parseComplexType(i)
+	var value interface{}
+	var err error
+
+	arraySize := int(C.GetArraySize(p.record, p.info, C.int(i)))
+	result := make([]interface{}, arraySize)
+
+	for j := 0; j < arraySize; j++ {
+		if int(C.PropertyIsStruct(p.info, C.int(i))) == 1 {
+			value, err = p.parseComplexType(i)
+		} else {
+			value, err = p.parseSimpleType(i)
+		}
+		if err != nil {
+			return nil, err
+		}
+		result[j] = value
 	}
-	return p.parseSimpleType(i)
+
+	return result, nil
 }
```

Comment:
Didn't catch the idea. Why we return an array on any case? 

It seems pretty complex to use. Could we detect a real arrays using some API and return other fields as values?

Reply:
Now it returns a value if array length is 1.